app/src/main/kotlin/com/hippo/ehviewer/ui/legacy/FabLayout.kt

```diff
@@ -1,482 +1,138 @@
-                        if (child.visibility == GONE) {
-                            continue
-                        }
-                        child.visibility = if (expanded) VISIBLE else INVISIBLE
-                        if (expanded) {
-                            child.alpha = 1f
-                        }
-                    }
-                } else {
-                    if (mHidePrimaryFab) {
-                        setPrimaryFabAnimation(getChildAt(count - 1), expanded, !expanded)
+            val coroutineScope = rememberCoroutineScope()
+            val appearState by animateFloatAsState(
+                targetValue = if (hidden) 0f else 1f,
+                animationSpec = tween(FAB_ANIMATE_TIME),
+                label = "hiddenState",
+            )
+            Box(
+                modifier = Modifier.fillMaxSize().padding(16.dp),
+                contentAlignment = Alignment.BottomEnd,
+            ) {
+                Box(contentAlignment = Alignment.Center) {
+                    val animatedProgress by animateFloatMergePredictiveBackAsState(expandedBackingField)
+                    FloatingActionButton(
+                        onClick = { expanded = !expanded },
+                        modifier = Modifier.rotate(lerp(-90f, 0f, appearState)).scale(appearState),
+                    ) {
+                        Icon(
+                            imageVector = Icons.Default.Add,
+                            contentDescription = null,
+                            modifier = Modifier.rotate(lerp(135f, 0f, animatedProgress)),
+                        )
                     }
-                    for (i in 0 until count - 1) {
-                        val child = getChildAt(i)
-                        if (child.visibility == GONE) {
-                            continue
+                    secondaryFab?.run {
+                        forEachIndexed { index, (imageVector, onClick) ->
+                            SmallFloatingActionButton(
+                                onClick = {
+                                    coroutineScope.launchIO {
+                                        runSuspendCatching {
+                                            onClick()
+                                        }
+                                        expanded = false
+                                    }
+                                },
+                                modifier = Modifier.layout { measurable, constraints ->
+                                    val placeable = measurable.measure(constraints)
+                                    layout(placeable.width, placeable.height) {
+                                        val distance = lerp(150 * (size - index) + 50, 0, animatedProgress)
+                                        placeable.placeRelative(0, -distance, -(size - index).toFloat())
+                                    }
+                                }.rotate(lerp(90f, 0f, appearState)).scale(appearState),
```

Comment:
Should secondary FAB rotate & scale? It's kind of strange.

Reply:
I think it's cool